homework/index.js

```diff
@@ -29,22 +29,97 @@
     return elem;
   }
 
+  function createTableRows(table, optionsHeader = {}, optionsValue = {}) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('td', tr, optionsHeader);
+    if (optionsHeader.text === 'Repository:') {
+      const td = createAndAppend('td', tr);
+      createAndAppend('a', td, optionsValue);
+    } else {
+      createAndAppend('td', tr, optionsValue);
+    }
+  }
+
+  function formatDate(dateString) {
+    const date = new Date(dateString);
+    let hours = date.getHours();
+    const ampm = hours >= 12 ? 'PM' : 'AM';
+    hours %= 12;
+    hours = hours === 0 ? 12 : hours;
+    let seconds = date.getSeconds();
+    seconds = seconds < 10 ? `0${seconds}` : seconds;
+    const formatedDate = `${date.getMonth() +
+      1}/${date.getDate()}/${date.getFullYear()}, ${hours}:${date.getMinutes()}:${seconds} ${ampm}`;
+    return formatedDate;
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const repoItem = createAndAppend('li', ul);
+    const table = createAndAppend('table', repoItem);
+    let description;
+    if (repo.description) {
+      description = repo.description;
+    } else {
+      description = 'No description';
+    }
+    const headerClass = 'td-header';
+    createTableRows(
+      table,
+      { text: 'Repository:', class: headerClass },
+      {
+        text: repo.name,
+        href: repo.html_url,
+        target: '_blank',
+      },
+    );
+    createTableRows(
+      table,
+      { text: 'Description:', class: headerClass },
+      { text: description },
+    );
+    createTableRows(
+      table,
+      { text: 'Forks:', class: headerClass },
+      { text: repo.forks },
+    );
+    createTableRows(
+      table,
+      { text: 'Updated', class: headerClass },
+      { text: formatDate(repo.updated_at) },
+    );
   }
 
   function main(url) {
     fetchJSON(url, (err, repos) => {
       const root = document.getElementById('root');
+      createAndAppend('div', root, {
+        class: 'main-header',
+        text: 'HYF-Repositories',
+      });
+
       if (err) {
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
         });
         return;
       }
+
       const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+
+      repos
+        .sort((repo, nextRepo) => {
```

Comment:
In the assignment we gave the hint to use `.localeCompare`. Your method works fine if all names just use the regular letters. However, as soon as they include letters with diacritics, such as in French, German, Dutch or Turkish the names are not sorted in the (dictionary) order you would expect.

You may want to try this code, comparing your sort method agains `.localeCompare()`. I have taken the surnames of the students in class23 but added another name, viz. **Öbul** which I would expect to be sorted just after **Obul**. With `.localeCompare()` (which takes into account the 'locale' that is set up in your computer) this works fine. Your sort method however places **Öbul** after **Zengin**.

```js
'use strict';

{
  const users = [
    'Hüsrev',
    'Raad',
    'Kanatsiz',
    'Ekinci',
    'Ince',
    'Ajaj',
    'Obul',
    'Öbul',
    'Zengin',
    'Karatas',
    'Ersaru',
    'Tercan',
    'Alnakib',
    'Grabetskaia',
    'Sönmez',
    'Akkuz',
  ];

  const esraSorted = users.slice().sort((a, b) => {
    const aUpper = a.toUpperCase();
    const bUpper = b.toUpperCase();
    if (aUpper < bUpper) {
      return -1;
    }
    if (aUpper > bUpper) {
      return 1;
    }
    return 0;
  });

  const localeSorted = users.slice().sort((a, b) => a.localeCompare(b));

  console.log(esraSorted);
  console.log(localeSorted);
}
```

Reply:
I just realized I missed the tip and thank you very much for explaining why my function is not very useful. I will try to be more careful reading instructions and writing better functions. Many thanks.